packages/poi/lib/server.js

```diff
@@ -8,11 +9,14 @@ module.exports = function (compiler, options = {}) {
   const port = options.port
   const host = options.host
 
-  const devMiddleWare = require('webpack-dev-middleware')(compiler, {
-    quiet: true,
-    publicPath: compiler.options.output.publicPath,
-    path: `http://${host}:${port}/__webpack_hmr`
-  })
+  const devMiddleWare = require('webpack-dev-middleware')(compiler, merge(
+    options.devMiddleware || {},
+    {
+      quiet: true,
+      publicPath: compiler.options.output.publicPath,
+      path: `http://${host}:${port}/__webpack_hmr`
+    }
+  ))
 
```

Comment:
maybe reverse the order so that `options.devServer` can override default values.